src/Stratis.Bitcoin.Features.MemoryPool.Tests/MemoryPoolTransactionTests.cs

```diff
@@ -337,7 +337,7 @@ public void MempoolAncestorIndexingTest()
             // tx1 and tx5 are both 10000
             // Ties are broken by hash, not timestamp, so determine which
             // hash comes first.
-            if (tx1.GetHash() < tx5.GetHash())
+            if (tx1.GetHash() > tx5.GetHash())
```

Comment:
Why did you have to change the test if the functionality did not change?

Reply:
It's explained in my top comment.

Reply:
Ah I see this is only for hashes ascending of ancestors.
But why change that? if it was like that carried over from bitcoin there may have been a good a reason.

Reply:
The Stratis implementation of TxMempoolEntry is sorted descending by fee but sorted ascending based on hash when the fees were equal by virtue of using **OrderBy** on:

```
if (f1 == f2)	
{	
     if (a.TransactionHash < b.TransactionHash) // Standard way of comparing values
           return -1;	
     return 1;	
}	

if (f1 > f2)    // Descending ordering on fees
     return -1;	
return 1;
```

For TxMemPoolModifiedEntry the behavior is different with a descending sort on both fees and hashes by virtue of the use of **OrderByDescending** on:
```
if (f1 == f2)
{	
   if (a.TransactionHash == b.TransactionHash) return 0;	
   if (a.TransactionHash < b.TransactionHash) return -1; // standard way of comparing values
   return 1;
}
	
return f1 > f2 ? 1 : -1; // Ascending ordering on fees. OrderByDescending compensating for this but then also flipping the hash comparison..!
```

Should I assume our legacy TxMemPoolModifiedEntry implementation is incorrect?

Reply:
Reverted the test case and made code changes based on the assumption that TxMemPoolModifiedEntry rather then TxMempoolEntry had an incorrectly implemented hash comparison.

Reply:
Yeah I took the tests from the bitcoin code so assume the tests are correct